.github/workflows/build.yml

```diff
@@ -1,22 +1,47 @@
 name: "build"
 on:
+  # XXX: Temp change
+  pull_request:
   push:
     branches:
       - master
 
 jobs:
-  build:
+  # XXX: Create and publish the builder image separatedly
+  docker:
     runs-on: ubuntu-latest
     steps:
     - name: Checkout repo
       uses: actions/checkout@v2
       with:
+        # XXX: Why is this necessary?
         ref: ${{ github.head_ref }}
 
-    - name: Docker
-      uses: docker/build-push-action@v1
-      with:
-        username: ${{ secrets.DOCKER_USERNAME }}
-        password: ${{ secrets.DOCKER_PASSWORD }}
-        repository: sentryintegrations/sentry-github-action-release
-        tags: latest
+    - name: Registry login
+      run: |
+        echo ${{ secrets.GITHUB_TOKEN }} | docker login ghcr.io -u $GITHUB_ACTOR --password-stdin
+
+    # BUILDKIT_INLINE_CACHE creates the image in such a way that you can
+    # then use --cache-from (think of a remote cache)
+    # This feature can be used because of using the buildx plugin
+    # In the case of the Docker builder iamge, we should always see that it's using the remote cache
+    # XXX: Would a Docker push notice that is already there?
+    - name: Docker build images
+      run: |
+        docker buildx build . \
+          --build-arg BUILDKIT_INLINE_CACHE=1 \
+          -t ghcr.io/getsentry/action-release-image:latest \
+          -t ghcr.io/getsentry/action-release-image:${{ github.sha }} \
+          --cache-from ghcr.io/getsentry/action-release-image:latest \
+          --target app
```

Comment:
In a previous commit I was building both the builder and the app and publishing both, however, since the builder image is always being invalidated I will just build the app target.

Reply:
oh hmm, you could probably push both, then pull both, then use two `--cache-from` arguments to make it not invalidated each time?